app/src/common/shared/com/igalia/wolvic/ui/widgets/NavigationBarWidget.java

```diff
@@ -773,6 +778,14 @@ private void enterFullScreenMode() {
         mWidgetManager.pushWorldBrightness(mBrightnessWidget, mBrightnessWidget.getSelectedBrightness());
     }
 
+    private void disableHeadlockForVRVideo() {
+        SettingsStore settingsStore = SettingsStore.getInstance(getContext());
+        if (settingsStore.isHeadLockEnabled()) {
+            settingsStore.setHeadLockEnabled(false);
+            shouldRestoreHeadLockOnVRVideoExit = true;
+        }
+    }
```

Comment:
We can do it better. The way this is implemented is a bit risky because adding a new call to `enterVRVideo` will force us to add also a call to disable headlock, and it's very likely that we could forget about that in some months. So the same way we restore headlock in `exitVRVideo` we can get rid of this new private method and all the code to the end of `enterVRVideo`. This way we'll we safe against future calls.

Reply:
`enterVRVideo` calls `showVRVideo`, and in some places we use `showVRVideo` too. So I put this at the end of `showVRVideo`.